File: samples/server/petstore/fsharp-giraffe/OpenAPI/src/impl/UserApiService.fs
namespace OpenAPI
open OpenAPI.Model.User
open UserApiHandlerParams
open UserApiServiceInterface
open System.Collections.Generic
open System
open Giraffe

module UserApiServiceImplementation =
    
    //#region Service implementation
    type UserApiServiceImpl() = 
      interface IUserApiService with
      
        member this.CreateUser ctx args =
            let content = "successful operation" 
            CreateUserDefaultStatusCode { content = content }

        member this.CreateUsersWithArrayInput ctx args =
            let content = "successful operation" 
            CreateUsersWithArrayInputDefaultStatusCode { content = content }

        member this.CreateUsersWithListInput ctx args =
            let content = "successful operation" 
            CreateUsersWithListInputDefaultStatusCode { content = content }

        member this.DeleteUser ctx args =
          if true then 
            let content = "Invalid username supplied" 
            DeleteUserStatusCode400 { content = content }
          else
            let content = "User not found" 
            DeleteUserStatusCode404 { content = content }

        member this.GetUserByName ctx args =
          if true then 
            let content = "successful operation" :> obj :?> User // this cast is obviously wrong, and is only intended to allow generated project to compile   
            GetUserByNameStatusCode200 { content = content }
          else if true then 
            let content = "Invalid username supplied" 
            GetUserByNameStatusCode400 { content = content }
          else
            let content = "User not found" 
            GetUserByNameStatusCode404 { content = content }

        member this.LoginUser ctx args =
          if true then 
            let content = "successful operation" :> obj :?> string // this cast is obviously wrong, and is only intended to allow generated project to compile   
            LoginUserStatusCode200 { content = content }
          else
            let content = "Invalid username/password supplied" 
            LoginUserStatusCode400 { content = content }

        member this.LogoutUser ctx  =
            let content = "successful operation" 
            LogoutUserDefaultStatusCode { content = content }

        member this.UpdateUser ctx args =
          if true then 
            let content = "Invalid user supplied" 
            UpdateUserStatusCode400 { content = content }
          else
            let content = "User not found" 
            UpdateUserStatusCode404 { content = content }

      //#endregion

    let UserApiService = UserApiServiceImpl() :> IUserApiService

File: samples/server/petstore/kotlin-springboot-x-kotlin-implements/src/main/kotlin/org/openapitools/api/UserApiService.kt
package org.openapitools.api

import org.openapitools.model.User

interface UserApiService {

    /**
     * POST /user : Create user
     *
     * @param user  (required)
     * @return successful operation (status code 200)
     * @see UserApi#createUser
     */
    fun createUser(user: User): Unit

    /**
     * POST /user/createWithArray : Creates list of users with given input array
     *
     * @param user  (required)
     * @return successful operation (status code 200)
     * @see UserApi#createUsersWithArrayInput
     */
    fun createUsersWithArrayInput(user: kotlin.collections.List<User>): Unit

    /**
     * POST /user/createWithList : Creates list of users with given input array
     *
     * @param user  (required)
     * @return successful operation (status code 200)
     * @see UserApi#createUsersWithListInput
     */
    fun createUsersWithListInput(user: kotlin.collections.List<User>): Unit

    /**
     * DELETE /user/{username} : Delete user
     *
     * @param username  (required)
     * @return Invalid username supplied (status code 400)
     *         or User not found (status code 404)
     * @see UserApi#deleteUser
     */
    fun deleteUser(username: kotlin.String): Unit

    /**
     * GET /user/{username} : Get user by user name
     *
     * @param username  (required)
     * @return successful operation (status code 200)
     *         or Invalid username supplied (status code 400)
     *         or User not found (status code 404)
     * @see UserApi#getUserByName
     */
    fun getUserByName(username: kotlin.String): User

    /**
     * GET /user/login : Logs user into the system
     *
     * @param username  (required)
     * @param password  (required)
     * @return successful operation (status code 200)
     *         or Invalid username/password supplied (status code 400)
     * @see UserApi#loginUser
     */
    fun loginUser(username: kotlin.String, password: kotlin.String): kotlin.String

    /**
     * GET /user/logout : Logs out current logged in user session
     *
     * @return successful operation (status code 200)
     * @see UserApi#logoutUser
     */
    fun logoutUser(): Unit

    /**
     * PUT /user/{username} : Updated user
     *
     * @param username  (required)
     * @param user  (required)
     * @return Invalid user supplied (status code 400)
     *         or User not found (status code 404)
     * @see UserApi#updateUser
     */
    fun updateUser(username: kotlin.String, user: User): Unit
}
